add breaking change for the `Story` type change (#19011)
storybookjs/storybook: MIGRATION.md

### Breaking changes

#### `Story` type change to `StoryFn`, and the new `Story` type now refers to `StoryObj`

In 6.x you were able to do this:

```js
import type { Story } from '@storybook/react';

export const MyStory: Story = () => <div />;
```

But this will produce an error in 7.0 because `Story` is now a type that refers to the `StoryObj` type.
You must now use the new `StoryFn` type:

```js
import type { StoryFn } from '@storybook/react';

export const MyStory: StoryFn = () => <div />;
```

> This change was done to improve the experience of writing CSF3 stories, which the recommended way of writing stories in 7.0.:

Reviewer: ```suggestion
This change was done to improve the experience of writing CSF3 stories, which is the recommended way of writing stories in 7.0:
```

```js
import type { Story } from '@storybook/react';
import { Button, ButtonProps } from './Button';

export default {
  component: Button,
};

export const Primary: Story<ButtonProps> = {
  variant: 'primary',
};
```

> If want to be explicit, you can also import `StoryObj` instead of `Story`, they are the same type.

Reviewer: ```suggestion
If you want to be explicit, you can also import `StoryObj` instead of `Story`, they are the same type.
```

### Breaking changes

#### `Story` type change to `StoryFn`, and the new `Story` type now refers to `StoryObj`

> In 6.x you were able to do this:

Reviewer: ```suggestion
In 6.x, the default type for Stories related to the **story function** syntax, such as:
```

```js
import type { Story } from '@storybook/react';

export const MyStory: Story = () => <div />;
```

But this will produce an error in 7.0 because `Story` is now a type that refers to the `StoryObj` type.
You must now use the new `StoryFn` type:

```js
import type { StoryFn } from '@storybook/react';

export const MyStory: StoryFn = () => <div />;
```

This change was done to improve the experience of writing CSF3 stories, which the recommended way of writing stories in 7.0.:

```js
import type { Story } from '@storybook/react';
import { Button, ButtonProps } from './Button';

export default {
  component: Button,
};

> 

Reviewer: ```suggestion
```
